doc/proposals/manage-dockerfiles.md

```diff
@@ -0,0 +1,24 @@
+## Manage Dockerfiles
+
+### Goal
+The Operator-SDK should allow users to create multiple images with different base images easily. We want the users to be able to decide dynamically choose the base image. The Operator-SDK should be able to change and own the steps that it needs to take for whichever type of operator. 
+
+### Non-Goals
+* Allow users to change the flow of how images are created
+* Support users to add actions/steps to the end of the image creation.
+
+### Proposed Solution
+
+#### User-facing changes
+The `operator-sdk build` command will add an option to pass in the base image example: `operator-sdk build --base-dockerfile centos-base.Dockerfile`. The `new` command will generate a default base, as it does now, called `base.Dockerfile`. If no option is passed in `base.Dockerfile`  will be selected.
+
+#### Operator-SDK Changes
+The `operator-sdk` will now add an `sdk.Dockerfile`. This dockerfile will contain all the steps to set up the operator image correctly for whichever type of operator this is. This will be changed/updated and owned by the operator-sdk, and we will not expect or respect any changes made by a user to this file. This will allow us to completely re-write it as we see fit. The testing framework will not be changed as part of this proposal.
```

Comment:
Just a thought but how important is it to name the top level Dockerfile `sdk.Dockerfile`. Wondering if we can keep this as `build/Dockerfile` so it doesn't break users with older projects.
I know it's a small change to update existing projects to move `build/Dockerfile` to `build/sdk.Dockerfile` but it would be nice if we could avoid even that if the name change isn't strictly necessary.